pallets/msa/README.md

```diff
@@ -0,0 +1,71 @@
+# MSA Pallet
+
+The MSA Pallet provides functionality for handling Message Source Accounts.
+
+## Summary
+
+The Message Source Account (MSA) is the primary user account system on Frequency for Messages and Stateful Storage.
+All users on Frequency must have an MSA in order to:
+
+1. Acquire a User Handle
+2. Delegate tasks to Providers (defining specific tasks for specific providers)
+3. Become a Provider so they may do Provider level tasks on their own behalf
+4. Have Message or Stateful Storage data
+
+### MSA Id and Keys
+
+Once a user creates an MSA, they are assigned an MSA Id, a unique number the time of creation with one or more keys attached for control.
+(A control key may only be attached to ONE MSA at any single point in time.)
+
+### Actions
+
+The MSA pallet provides for:
+
+- Creating, reading, updating, and deleting operations for MSAs.
+- Managing delegation relationships for MSAs.
+- Managing keys associated with MSAs.
+
+## Interactions
+
+### Extrinsics
+
+| Name/Description                                                                              | Caller                                   | Payment            | Key Events                                                                                                                                                                                                                                       | Runtime Added |
+| --------------------------------------------------------------------------------------------- | ---------------------------------------- | ------------------ | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ | ------------- |
+| `add_public_key_to_msa`<br />Add MSA control key                                              | Delegator or Provider with Signature     | Free               | [`PublicKeyAdded`](https://frequency-chain.github.io/frequency/pallet_msa/pallet/enum.Event.html#variant.PublicKeyAdded)                                                                                                                         | 1             |
```

Comment:
nit: Depending on the audience, is `Free` ambiguous? e.g. Free for end users, who are using Capacity. Not sure if this is a good spot to make the distinction.

Reply:
Actually this line is wrong! This one is supposed to be `Capacity` which is separate from `Free`. The `Caller` column was supposed to help disambiguate Capacity and free.